docs/user-authentication.txt

```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
 
+.. _laravel-user-auth-passport:
+
+Laravel Passport
+~~~~~~~~~~~~~~~~
+
+Laravel Passport is an OAuth 2.0 server implementation that offers
+API authentication for Laravel applications. Use Laravel Passport if
+your application requires OAuth2 support.
+
+Install Laravel Passport
+````````````````````````
+
+To install Laravel Passport and run the database migrations required
+to store OAuth2 clients, run the following command from your project root:
+
+.. code-block:: bash
+
+   php artisan install:api --passport
+
+Next, navigate to your ``User`` model and add the ``Laravel\Passport\HasApiTokens``
+trait. This trait provides helper methods that allow you to inspect a user's
+authentication token and scopes. The following code shows how to add ``Laravel\Passport\HasApiTokens``
+to your ``app\Models\User.php`` file:
+
+.. code-block:: php
+
+   <?php
+
+   namespace App\Models;
+
+   use MongoDB\Laravel\Auth\User as Authenticatable;
+   use Laravel\Passport\HasApiTokens;
+
+   class User extends Authenticatable
+   {
+      use HasApiTokens;
+      ...
+   }
+
+Then, define an ``api`` authentication guard in your ``config\auth.php``
+file and set the ``driver`` option to ``passport``. This instructs your
+application to use Laravel Passport's ``TokenGuard`` class to authenticate
+API requests. The following example adds the ``api`` authentication guard
+to the ``guards`` array:
+
+.. code-block:: php
+   :emphasize-lines: 7-10
+
+   'guards' => [
+      'web' => [
+         'driver' => 'session',
+         'provider' => 'users',
+      ],
+   
+      'api' => [
+         'driver' => 'passport',
+         'provider' => 'users',
+      ],
+   ],
+
+Use Laravel Passport with Laravel MongoDB
+`````````````````````````````````````````
+
+After installing Laravel Passport, you must enable Passport compatibility with MongoDB by
+defining custom {+odm-short+} models that extend the corresponding Passport models.
+To extend each Passport model class, include the ``DocumentModel`` trait in the custom models.
+You can define the following {+odm-short+} model classes:
+
+- ``MongoDB\Laravel\Passport\AuthCode``, which extends ``Laravel\Passport\AuthCode``
+- ``MongoDB\Laravel\Passport\Client``, which extends ``Laravel\Passport\Client``
+- ``MongoDB\Laravel\Passport\PersonalAccessClient``, which extends ``Laravel\Passport\PersonalAccessClient``
+- ``MongoDB\Laravel\Passport\RefreshToken``, which extends ``Laravel\Passport\RefreshToken``
+- ``MongoDB\Laravel\Passport\Token``, which extends ``Laravel\Passport\Token``
+
+For example, the following code overrides the default ``Laravel\Passport\AuthCode``
+model class by defining a ``MongoDB\Laravel\Passport\AuthCode`` class and including
+the ``DocumentModel`` trait:
+
+.. code-block:: php
+
+   class MongoDB\Laravel\Passport\AuthCode extends Laravel\Passport\AuthCode
+   {
+      use MongoDB\Laravel\Eloquent\DocumentModel;
+
+      protected $primaryKey = '_id';
+      protected $keyType = 'string';
+   }
+
```

Comment:
Q: does the user need to create models for each of the listed classes? If so, that should be more clear in a note or something similar.

Reply:
Yeah each one needs a custom model. I changed the wording of the paragraph below (line 246) to clarify that, lmk if it's clear enough